express-post-json/index.js

```diff
@@ -0,0 +1,29 @@
+const express = require('express');
+
+const app = express();
+
+const grades = [];
+
+let nextId = 1;
+
+const jsonExpress = express.json();
+
+app.get('/api/grades', (req, res) => {
+  res.json(grades);
+});
+
+app.use(jsonExpress);
+
+app.post('/api/grades', (req, res) => {
+  const newGrade = req.body;
+  newGrade.id = nextId;
+  nextId++;
+  grades.push(newGrade);
+  res.json(newGrade);
+  res.status(201);
```

Comment:
Be sure to set the status code before sending the body. The response message is written line-by-line from the top, so as soon as you do anything that sends or starts sending the body, you can't modify the status code or headers anymore.